### PR TITLE
inventory collection apis and tests

### PR DESCRIPTION
**Pull request by vvb**
Signed-off-by: vvb <vvb@cisco.com>

### CODE REVIEW

File: imcsdk/apis/server/inventory.py [hunk 1]
```diff
@@ -11,560 +11,306 @@
...
+
+    html += "</table>"
+    html += "</body>"
+    html += "</html>"
+    f.write(html)
+    f.close()
+
+
+def get_inventory(handle,
+                  component="all",
```
Comment (waghswapnil) [inline imcsdk/apis/server/inventory.py:316]: Should we also handle component as a list of specific components like 
`component = ['storage', 'disks']`

File: imcsdk/apis/server/inventory.py [hunk 2]
```diff
@@ -11,560 +11,306 @@
...
-            print(callback_dict.keys())
+        servers = [handle]
+
+    for server in servers:
+        if component == "all":
+            for comp in spec.keys():
+                _get_inventory(server, comp, spec, inventory)
+        else:
+            if component not in spec:
+                raise "Unsupported component type" + str(component)
```
Comment (waghswapnil) [inline imcsdk/apis/server/inventory.py]: minor correction:-
`raise ImcOperationError("Inventory Collection", "Unsupported component type: %s" % component)`

File: imcsdk/apis/server/inventory.py [hunk 3]
```diff
@@ -11,560 +11,306 @@
...
+    import csv
+    if file_name is None:
+        raise ImcOperationError("Inventory collection",
+                                "file_name is a required parameter")
+    f = csv.writer(open(file_name, "wb"))
+
+    x = inventory
+    for comp in spec:
+        f.writerow([comp.upper()])
+        props = spec[comp]["props"]
```
Comment (waghswapnil) [inline imcsdk/apis/server/inventory.py]: => `comp["props"]`

File: imcsdk/apis/server/inventory.py [hunk 4]
```diff
@@ -11,560 +11,306 @@
...
+
+    html += "<head>"
+    html += "</head>"
+    html += "<body>"
+
+    x = inventory
+    for comp in spec:
+        html += '<table border="1">' + comp.upper()
+
+        props = spec[comp]["props"]
```
Comment (waghswapnil) [inline imcsdk/apis/server/inventory.py]: => `comp["props"]`

File: imcsdk/apis/server/inventory.py (hunk 2 above)
Comment (waghswapnil) [reply imcsdk/apis/server/inventory.py]: `raise 'str' ` raises another exception

```
raise "xyz"
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-9-fc992fe50778> in <module>()
----> 1 raise "xyz"

TypeError: exceptions must be old-style classes or derived from BaseException, not str
```

File: tests/server/test_inventory.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    get_inventory(handle, component="cpu", file_format="csv",
+                  file_name="inventory.csv")
+    get_inventory(handle, component="disks", file_format="json",
+                  file_name="inventory.json")
+    get_inventory(handle, component="vic")
+
+
+@raises(Exception)
+def test_get_inventory_error():
+    get_inventory(handle, component="vic", file_format="csv")
```
Comment (waghswapnil) [inline tests/server/test_inventory.py:54]: Can we add a test case for an invalid component ?

File: imcsdk/apis/server/inventory.py (hunk 2 above)
Comment (vvb) [reply imcsdk/apis/server/inventory.py]: already changed in the updated diff

File: tests/server/test_inventory.py (hunk 5 above)
Comment (vvb) [reply tests/server/test_inventory.py:54]: added

File: imcsdk/apis/server/inventory.py (hunk 1 above)
Comment (vvb) [reply imcsdk/apis/server/inventory.py:316]: done

**Review by waghswapnil — APPROVED** [review]
LGTM

**Pull request merged**